edgelet/edgelet-docker/src/runtime.rs

```diff
@@ -451,44 +484,32 @@ impl ModuleRuntime for DockerModuleRuntime {
             return Box::new(future::err(Error::from(err)));
         }
 
-        let (sender, receiver): (Sender<()>, Receiver<()>) = oneshot::channel();
-
-        if let Err(err) = self
-            .create_socket_channel
-            .unbounded_send(ModuleAction::Start(id.clone(), sender))
-            .map_err(|_| {
-                Error::from(ErrorKind::RuntimeOperation(RuntimeOperation::GetModule(
-                    id.clone(),
-                )))
-            })
-        {
-            return Box::new(future::err(err));
-        }
-
-        let future = self.client.container_api().container_start(&id, "");
-
-        let id2 = id.clone();
-
+        let socket_signal = self.create_socket_channel.clone();
         Box::new(
-            receiver
-                .map_err(move |_| {
-                    Error::from(ErrorKind::RuntimeOperation(RuntimeOperation::GetModule(id)))
-                })
-                .and_then(move |()| {
-                    future.then(move |result| match result {
-                        Ok(_) => {
-                            info!("Successfully started module {}", id2);
-                            Ok(())
-                        }
-                        Err(err) => {
-                            let err = Error::from_docker_error(
-                                err,
-                                ErrorKind::RuntimeOperation(RuntimeOperation::StartModule(id2)),
-                            );
-                            log_failure(Level::Warn, &err);
-                            Err(err)
+            self.client
+                .container_api()
+                .container_start(&id, "")
+                .then(|result| match result {
+                    Ok(_) => {
+                        info!("Successfully started module {}", id);
+                        Ok(())
+                    }
+                    Err(err) => {
+                        let id = id;
+                        let socket_signal = socket_signal;
```

Comment:
Remove these and make this closure a `move` closure instead.

Reply:
Fixed